renderer/components/Onboarding/Steps/Recover.js

```diff
@@ -131,40 +345,42 @@ class Recover extends React.Component {
         }}
         onSubmitFailure={onSubmitFailure}
       >
-        <Header
-          subtitle={<FormattedMessage {...messages.import_description} />}
-          title={<FormattedMessage {...messages.import_title} />}
-        />
-
-        <Bar my={4} />
-
-        <Flex justifyContent="space-between">
-          <Flex as="ul" flexDirection="column" mr={2}>
-            {indexes.slice(0, 6).map(word => (
-              <SeedWordWithIntl
-                key={word + 1}
-                index={word + 1}
-                onPaste={word === 0 ? this.handlePaste : null}
-                word={seed[word]}
-              />
-            ))}
-          </Flex>
-          <Flex as="ul" flexDirection="column" mx={2}>
-            {indexes.slice(6, 12).map(word => (
-              <SeedWordWithIntl key={word + 1} index={word + 1} word={seed[word]} />
-            ))}
-          </Flex>
-          <Flex as="ul" flexDirection="column" mx={2}>
-            {indexes.slice(12, 18).map(word => (
-              <SeedWordWithIntl key={word + 1} index={word + 1} word={seed[word]} />
-            ))}
-          </Flex>
-          <Flex as="ul" flexDirection="column" ml={2}>
-            {indexes.slice(18, 24).map(word => (
-              <SeedWordWithIntl key={word + 1} index={word + 1} word={seed[word]} />
-            ))}
-          </Flex>
-        </Flex>
+        {({ formState }) => (
+          <>
+            <Header
+              subtitle={<FormattedMessage {...messages.import_description} />}
+              title={<FormattedMessage {...messages.import_title} />}
+            />
+
+            <Bar my={4} />
+
+            <Flex justifyContent="space-between">
+              {[[0, 6], [6, 12], [12, 18], [18, 24]].map((slice, sliceIndex) => (
+                <Flex
+                  key={sliceIndex}
+                  as="ul"
+                  flexDirection="column"
+                  ml={sliceIndex > 0 ? 2 : 0}
+                  mr={sliceIndex < 3 ? 2 : 0}
+                >
+                  {indexes.slice(...slice).map(index => (
+                    <SeedWordWithIntl
+                      key={index + 1}
+                      bip39WordMap={this.bip39WordMap}
+                      formState={formState}
+                      index={index + 1}
+                      onPaste={index === 0 ? this.handlePaste : null}
+                      setFormApiFieldValue={word => {
+                        this.setFormApiFieldValue(index)(word)
```

Comment:
I did so that the partially applied function can be called like this `setFormApiFieldValue(selection)` in `SeedWord`. I guess it's not really needed since we pass the index to `SeedWord` and can just call the function like this `setFormApiFieldValue(index - 1, selection)` but it doesn't look as nice?

What do you think? Remove curried function?

Reply:
Kept the curry for now unless we really wanna remove it.